### ISSUE TITLE
feat: Common DA Trait

### ISSUE THREAD

**Issue by drspacemn** [keep-starknet-strange/madara#983]
There are currently 3 DA implementations being pushed in tandem:
- Avail -> https://github.com/Leouarz/madara/pull/1/files
- Celestia ->  https://github.com/keep-starknet-strange/madara/pull/967/files
- Ethereum -> https://github.com/keep-starknet-strange/madara/tree/pair-programming/exploration-team

The goal is to have a common trait that each of these DA solutions can implement whereby the DA worker can run depending on what DA solutions the node has been initialized with:
- `::publish_state_diff`
- `::verify_inclusion`
- `::<possibly others>`

 This will also include a method to distinguish what DA solution the node has be initialized with:
 
 - `--da [ethereum|celestia|avail]`
 - `--da-node <Node Address>`

**Comment by bbrandtom** [comment]
One thing that is missing is how this will be verified on L1.
For Ethereum this already exist in the current implementation of L1 core contract of starknet, but I’m not sure how this will happen with Celestial and Avail. Do you have any ideas on that.

**Comment by abdelhamidbakhta** [comment]
> 

Hey @bbrandtom , good question. For Celestia it can be done through the Quantum Gravity Bridge (https://blog.celestia.org/celestiums/)

![celestia_infograf_1 _feb9](https://github.com/keep-starknet-strange/madara/assets/45264458/9cce68af-b4d3-492e-a4da-dd26365e511d)

TL;DR:
```
Celestia's data availability bridge contract, the Quantum Gravity Bridge, will sit on Ethereum. Ethereum L2 operators will be able to post their transaction data to the Celestia network, where it is placed into blocks by Celestia's Proof of Stake (PoS) validators. This data is then relayed in the form of a data availability attestation from Celestia to Ethereum. The attestation is a Merkle root of the L2 data signed by the Celestia validators attesting to the fact that the data is available on Celestia.
```

I don't know about Avail but I guess there is a similar mechanism. @MdTeach can you confirm ?

**Comment by MdTeach** [comment]
@abdelhamidbakhta i think you have mistaken me for the tg user @asyncakash

**Comment by anurag-arjun** [comment]
Yes there is a data attestation bridge from Avail to Ethereum - https://blog.availproject.org/data-attestation-bridge/. @abdelhamidbakhta @bbrandtom

**Comment by abdelhamidbakhta** [comment]
> @abdelhamidbakhta i think you have mistaken me for the tg user @asyncakash

oops sorry!

**Comment by tdelabro** [comment]
It has been implemented
